image/src/app/module/main.py

```diff
@@ -17,6 +36,35 @@ def module_main(data):
         [string, string]: [data, error]
     """
     try:
-        return data, None
+        # build return body
+        if type(data) == dict:
+            return_body = processData(data)
+        else:
+            return_body = []
+            for data_instance in data:
+                return_body.append(processData(data_instance))
+
+        # publish data to a remote MQTT broker
+        result = client.publish(topic=APPLICATION['TOPIC'], payload=json.dumps(return_body), qos=APPLICATION['QOS'])
+
+        # result: [0, 1]
+        if result[0] == 0:
```

Comment:
From the function docs:
Returns a MQTTMessageInfo class, which can be used to determine whether the message has been delivered (using info.is_published()) or to block waiting for the message to be delivered (info.wait_for_publish()). The message ID and return code of the publish() call can be found at info.mid and info.rc.

For backwards compatibility, the MQTTMessageInfo class is iterable so the old construct of (rc, mid) = client.publish(...) is still valid.

rc is MQTT_ERR_SUCCESS to indicate success or MQTT_ERR_NO_CONN if the client is not currently connected. mid is the message ID for the publish request. The mid value can be used to track the publish request by checking against the mid argument in the on_publish() callback if it is defined.

Reply:
Please use either `MQTTMessageInfo` class and its methods as described in the first paragraph or `(rc, mid)` tuple as described in the second.

Reply:
I implemented `(rc,mid)` method, but we use only `rc` value, because we don't use on_publish() callback.

Reply:
you don't need the parenthesis around `rc, mid`

Reply:
please also try to use named constants instead of some numbers. in this case `MQTT_ERR_SUCCESS`

Reply:
Done